dojo/tools/zap/parser.py

```diff
@@ -88,13 +57,16 @@ def get_items(self, tree, test):
 
                 find.unsaved_endpoints = [main_host]
                 for i in item.items:
-                    parts = urlparse(i['uri'])
-                    find.unsaved_endpoints.append(Endpoint(protocol=parts.scheme,
-                                                           host=parts.netloc[:500],
-                                                           path=parts.path[:500],
-                                                           query=parts.query[:1000],
-                                                           fragment=parts.fragment[:500],
-                                                           product=test.engagement.product))
```

Comment:
i remember seeing some very long urls overflowing the max length of some fields, so maybe these truncate actions must remain, or maybe even centralized in EndPoint.save() or something that works nice with get_or_create

Reply:
yeah. I was wondering if I could just fix the issue and wait for the big PR that clean the endpoint mess before improve this part. Will try that tomorrow.